File: chapter2/python/selection_sort.py
def selection_sort(arr: []) -> []:
    sorted_arr = []
    l = len(arr)
    for _ in range(l):
        m = max(arr)
        sorted_arr.append(m)
        arr.remove(m)
    return sorted_arr


if __name__ == "__main__":
    print(selection_sort([4,1,3,2]))

File: selection-sort/go/selection_sort.go
package selectionsort

import (
	helper "grokking-algorithms/go-helper"
)

func SelectionSort(arr []int) []int {
	sortedArr := []int{}
	l := len(arr)
	for i := 0; i < l; i++ {
		maxIndex, max := helper.Max(arr)
		sortedArr = append(sortedArr, max)
		arr = helper.DeleteByIndex(arr, maxIndex)
	}
	return sortedArr
}
